hasher.go

```diff
@@ -274,14 +283,10 @@ func (n *Hasher) HashNode(left, right []byte) ([]byte, error) {
 	leftMinNs, leftMaxNs := left[:n.NamespaceLen], left[n.NamespaceLen:flagLen]
 	rightMinNs, rightMaxNs := right[:n.NamespaceLen], right[n.NamespaceLen:flagLen]
 
-	minNs := min(leftMinNs, rightMinNs)
-	var maxNs []byte
-	if n.ignoreMaxNs && n.precomputedMaxNs.Equal(leftMinNs) {
-		maxNs = n.precomputedMaxNs
```

Comment:
If leftMinNs represents the maximum possible namespace ID, then by definition, rightMax must also be the maximum possible namespace ID. Therefore, this check appears to be unnecessary.